src/detector/implementations/usb_webcam_implementation.py

```diff
@@ -0,0 +1,202 @@
+from src.detector.api import \
+    GetCaptureDeviceResponse, \
+    GetCaptureImageRequest, \
+    GetCaptureImageResponse, \
+    GetCapturePropertiesResponse, \
+    SetCaptureDeviceRequest, \
+    SetCapturePropertiesRequest
+from src.common import \
+    EmptyResponse, \
+    ErrorResponse, \
+    get_kwarg, \
+    MCastResponse
+from src.common.structures.capture_status import CaptureStatus
+
+from src.detector.implementations import AbstractCameraInterface
+
+import base64
+import cv2
+import datetime
+import logging
+import numpy
+import os
+from typing import Any, Callable
+
+logger = logging.getLogger(__name__)
+
+# TODO: update type hints
+class USBWebcamWithOpenCV(AbstractCameraInterface):
```

Comment:
I will assume that this class is more or less verbatim for what existed previously, so I won't go through it in detail